packages/@aws-cdk/cli-lib/lib/cli.ts

```diff
@@ -0,0 +1,276 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { exec as runCli } from 'aws-cdk/lib';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { createAssembly, prepareContext, prepareDefaultEnvironment } from 'aws-cdk/lib/api/cxapp/exec';
+import { SharedOptions, DeployOptions, DestroyOptions, SynthOptions, ListOptions, StackActivityProgress } from './commands';
+
+/**
+ * AWS CDK CLI operations
+ */
+export interface IAwsCdkCli {
+  /**
+   * cdk list
+   */
+  list(options?: ListOptions): Promise<void>;
+
+  /**
+   * cdk synth
+   */
+  synth(options?: SynthOptions): Promise<void>;
+
+  /**
+   * cdk deploy
+   */
+  deploy(options?: DeployOptions): Promise<void>;
+
+  /**
+   * cdk destroy
+   */
+  destroy(options?: DestroyOptions): Promise<void>;
+}
+
+/**
+ * Configuration for creating a CLI from an AWS CDK App directory
+ */
+export interface CdkAppDirectoryProps {
+  /**
+   * Command-line for executing your app or a cloud assembly directory
+   * e.g. "node bin/my-app.js"
+   * or
+   * "cdk.out"
+   *
+   * @default - read from cdk.json
+   */
+  readonly app?: string
+
+  /**
+   * Emits the synthesized cloud assembly into a directory
+   *
+   * @default cdk.out
+   */
+  readonly output?: string;
+}
+
+/**
+ * A class returning the path to a Cloud Assembly Directory when its `produce` method is invoked with the current context
+ *
+ * AWS CDK apps might need to be synthesized multiple times with additional context values before they are ready.
+ * When running the CLI from inside a directory, this is implemented by invoking the app multiple times.
+ * Here the `produce()` method provides this multi-pass ability.
+ */
+export interface ICloudAssemblyDirectoryProducer {
+  /**
+   * The working directory used to run the Cloud Assembly from.
+   * This is were a `cdk.context.json` files will be written.
+   *
+   * @default - current working directory
+   */
+  workingDirectory?: string;
+
+  /**
+   * Synthesize a Cloud Assembly directory for a given context.
+   *
+   * For all features to work correctly, `cdk.App()` must be instantiated with the received context values in the method body.
+   * Usually obtained similar to this:
+   * ```ts fixture=imports
+   * class MyProducer implements ICloudAssemblyDirectoryProducer {
+   *   async produce(context: Record<string, any>) {
+   *     const app = new cdk.App({ context });
+   *     // create stacks here
+   *     return app.synth().directory;
+   *   }
+   * }
+   * ```
+   */
+  produce(context: Record<string, any>): Promise<string>
```

Comment:
No environment?

If this library is used via jsii, then the `process.env.CDK_DEFAULT_ACCOUNt = '1234';` statements will be executed in the Node process, which is separate from the JVM so `System.getEnvironment('CDK_DEFAULT_ACCOUNT')` will not find it.

Reply:
Well that sucks. Now I have to think gain. 😭